src/Generator/ShopzillaDE.php

```diff
@@ -41,115 +58,235 @@ public function __construct(ArrayHelper $arrayHelper)
+            'Whole file generation duration' => microtime(true) - $startTime,
+        ]);
     }
 
     /**
-     * @param RecordList $idlResultList
+     * Creates the header of the CSV file.
+     *
+     * @return array
      */
-    private function createIdlArray($idlResultList)
+    private function head():array
     {
-        if($idlResultList instanceof RecordList)
+        return array(
+            'Kategorie',
+            'Hersteller',
+            'Bezeichnung',
+            'Beschreibung',
+            'Artikel-URL',
+            'Bild-URL',
+            'SKU',
+            'Bestand',
+            'Versandgewicht',
+            'Zustand',
+            'Versandkosten',
+            'Gebot',
+            'Werbetext',
+            'EAN',
+            'Preis',
+            'Grundpreis',
+        );
+    }
+
+    /**
+     * Creates the variation row and prints it into the CSV file.
+     *
+     * @param array $variation
+     * @param KeyValue $settings
+     */
+    private function buildRow($variation, KeyValue $settings)
+    {
+        // Get the price list
+        $priceList = $this->elasticExportPriceHelper->getPriceList($variation, $settings);
+
+        // Only variations with the Retail Price greater than zero will be handled
+        if(!is_null($priceList['price']) && (float)$priceList['price'] > 0)
         {
-            foreach($idlResultList as $idlVariation)
-            {
-                if($idlVariation instanceof Record)
-                {
-                    $this->idlVariations[$idlVariation->variationBase->id] = [
-                        'itemBase.id' => $idlVariation->itemBase->id,
-                        'variationBase.id' => $idlVariation->variationBase->id,
-                        'variationRetailPrice.price' => $idlVariation->variationRetailPrice->price,
-                    ];
-                }
-            }
+            $data = [
+                'Kategorie'         => $this->elasticExportHelper->getCategory((int)$variation['data']['defaultCategories'][0]['id'], $settings->get('lang'), $settings->get('plentyId')),
+                'Hersteller'        => $this->elasticExportHelper->getExternalManufacturerName((int)$variation['data']['item']['manufacturer']['id']),
+                'Bezeichnung'       => $this->elasticExportHelper->getMutatedName($variation, $settings, 256),
+                'Beschreibung'      => $this->elasticExportHelper->getMutatedDescription($variation, $settings, 256),
+                'Artikel-URL'       => $this->elasticExportHelper->getMutatedUrl($variation, $settings, true, false),
+                'Bild-URL'          => $this->elasticExportHelper->getMainImage($variation, $settings),
+                'SKU'               => (string)$this->elasticExportHelper->generateSku($variation['id'], (float)$settings->get('referrerId'), 0, (string)$variation['data']['skus'][0]['sku']),
```

Comment:
If the customer has configured his export wrong regarding the referrer (old caption "all", now displayed as nothing, but value still -1), the SKU will be generated maybe by id -1 which produces an exception or shitty data. 
Let's please discuss together with Max what we want to do in case of formats, which have no referrer. Not generating a SKU? Set the variation ID maybe als default?